pkg/server/requestinfo/kcp_request_info_resolver.go

```diff
@@ -33,16 +34,28 @@ func NewKCPRequestInfoResolver() *KCPRequestInfoResolver {
 	}
 }
 
-var clustersRE = regexp.MustCompile(`/clusters/[^/]+/(.*)$`)
+var clustersRE = regexp.MustCompile(`/clusters/[^/]+(/.*)$`)
 
 func (k *KCPRequestInfoResolver) NewRequestInfo(req *http.Request) (*request.RequestInfo, error) {
-	matches := clustersRE.FindStringSubmatch(req.URL.Path)
-	if len(matches) == 2 {
-		// matches[0] is the leftmost pattern that matches (which includes /clusters/...) - skip over that
-		strippedURL := matches[1]
-		t := req.Clone(req.Context())
-		t.URL.Path = strippedURL
-		return k.delegate.NewRequestInfo(t)
+	if !containsPrefix([]string{"/apis/", "/api/"}, req.URL.Path) {
```

Comment:
with my comment [above](https://github.com/kcp-dev/kcp/pull/2961/files#r1228275536), this line could go and we would be back at the old code, right?